src/App.tsx

```diff
@@ -15,48 +22,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+function getSortGoods(
+  goods: string[],
+  sortField: SortType,
+  reverseField: boolean,
+) {
+  const listOfGoods = [...goods];
+
+  if (sortField) {
+    listOfGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.BY_ALPHABET:
+          return good1.localeCompare(good2);
+
+        case SortType.BY_LENGTH:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverseField) {
+    listOfGoods.reverse();
+  }
+
+  return listOfGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.BY_DEFAULT);
+  const [reverseField, setReverseField] = useState(false);
+
+  const visibleGoods = getSortGoods(
+    goodsFromServer,
+    sortField,
+    reverseField,
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          onClick={() => setSortField(SortType.BY_ALPHABET)}
```

Comment:
it is bad practise to make anonymous arrow function in jsx
consider to make function factory for both sort buttons
```
const makeSetSortField = (field) => () => setSortField(field);
```

```suggestion
          onClick={makeSetSortField(SortType.BY_ALPHABET)}
```